Formula/augustus.rb

```diff
@@ -22,54 +16,27 @@ class Augustus < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "8271784fc43729dd82e83e031ef63bb278771c6ba271ff7c7bc17908abc56646"
   end
 
-  # Fails to build with GCC 12
-  # https://github.com/Homebrew/homebrew-core/pull/106755
-  # https://github.com/Homebrew/homebrew-core/pull/40220
-  deprecate! date: "2022-08-03", because: :does_not_build
-
-  depends_on "boost" => :build
   depends_on "bamtools"
+  depends_on "boost"
+  depends_on "htslib"
+  depends_on "samtools"
 
   uses_from_macos "zlib"
 
-  on_macos do
-    depends_on "gcc"
-  end
-
   def install
-    # Avoid "fatal error: 'sam.h' file not found" by not building bam2wig
-    inreplace "auxprogs/Makefile", "cd bam2wig; make;", "#cd bam2wig; make;"
-
-    # Fix error: api/BamReader.h: No such file or directory
-    inreplace "auxprogs/bam2hints/Makefile",
-      "INCLUDES = /usr/include/bamtools",
-      "INCLUDES = #{Formula["bamtools"].include/"bamtools"}"
-    inreplace "auxprogs/filterBam/src/Makefile",
-      "BAMTOOLS = /usr/include/bamtools",
-      "BAMTOOLS= #{Formula["bamtools"].include/"bamtools"}"
-
-    # Prevent symlinking into /usr/local/bin/
-    inreplace "Makefile", %r{ln -sf.*/usr/local/bin/}, "#ln -sf"
-
     # Compile executables for macOS. Tarball ships with executables for Linux.
     system "make", "clean"
 
-    cd "src" do
-      if OS.mac?
-        # Clang breaks proteinprofile on macOS. This issue has been first reported
-        # to upstream in 2016 (see https://github.com/nextgenusfs/funannotate/issues/3).
-        # See also https://github.com/Gaius-Augustus/Augustus/issues/64
-        gcc_major_ver = Formula["gcc"].any_installed_version.major
-        with_env("HOMEBREW_CC" => Formula["gcc"].opt_bin/"gcc-#{gcc_major_ver}") do
-          system "make"
-        end
-      else
-        system "make"
-      end
-    end
+    system "make", "COMPGENEPRED=false",
```

Comment:
For simplicity, I just built without `COMPGENEPRED`. Can consider support later on by adding all dependencies (`gsl`, `suite-sparse`, `lp_solve`, `sqlite`, and `mysql++`).